File: sample/solution2.swift
class Solution {
      func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
          var seen = [Int: Int]()
          for (i, num) in nums.enumerated() {
              if let j = seen[target - num] {
                  return [j, i]
              }
              seen[num] = i
          }
          return []
      }
  }

File: sample/solution2.go
package main

type Solution2 struct{}

func (s *Solution) TwoSum2(nums []int, target int) []int {
	seen := make(map[int]int)
	for i, num := range nums {
		if j, ok := seen[target-num]; ok {
			return []int{j, i}
		}
		seen[num] = i
	}
	return []int{}
}
